### PR TITLE
RATEST-307: Delete EditDemographicsTest from the legacy selenium Tests

### PR DESCRIPTION
**Pull request by mherman22**
**Ticket ID** : https://issues.openmrs.org/browse/RATEST-307

**Objective**

- To Follow all the step definitions and scenarios given in https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature & Confirm that all test cases adhere with https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-legacy-tests/src/test/java/org/openmrs/contrib/qaframework/legacy/EditDemographicTest.java.

- There is a need to also see through this https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-legacy-tests/src/test/java/org/openmrs/contrib/qaframework/legacy/ContactInfoTest.java And confirm the tests are all covered with in this single workflow. If all the test cases are the same .Get rid of that classes, if not , lets update editDemographicsAndContactInfo workflow with the workflows remaining.

**Note**:

- I have gone through the above mentioned classes and updated the workflow with the cases from the legacy tests that were previously overlooked.
- Then i have deleted the legacy selenium tests

@sherrif10 @kdaud

### CODE REVIEW

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature [hunk 1]
```diff
@@ -1,21 +1,27 @@
...
-    When a user clicks on Edit link from Registration Summary Page
-    Then the system loads edit demographics section
+    Given a user clicks on the find patient app on the home page
+    Then the system loads the find patient page
+    And a user searches for the patient to edit
+    And a user clicks on the first patient
+    And a user clicks on Edit Registration Information link from Patient dashboard
+    Then the system loads Registration Summary Page
     And a user edits demographics
-    And a user clicks on the confirm button
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature:14]: any reason why the user story is being changed?

Comment (mherman22) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature:14]: Am trying as much as possible to follow the contactInfoTest and EditDemographicsTest selenium tests. Changing that user story is part of it.

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature:14]: According to this commit. Hopefully @kdaud  will find it more helpful concerning this comment [here](https://issues.openmrs.org/projects/RATEST/issues/RATEST-308?filter=reportedbyme)

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 2]
```diff
@@ -20,95 +24,132 @@
...
 
 public class EditDemographicsAndContactInfoSteps extends Steps {
 
+	private String prefix;
 	private TestData.PatientInfo testPatient;
+	private ActiveVisitsPage activeVisitsPage;
 	private RegistrationEditSectionPage registrationEditSectionPage;
-	private RegistrationSummaryPage registrationSummaryPage;
+	private FindPatientPage findPatientPage;
+	private static final String VISIT_TYPE_UUID = "7b0f5697-27e3-40c4-8bae-f4049abfb4ed";
```
Comment (sherrif10) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: Is this UUID  auto generated, How are you generating this uuid.

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature (hunk 1 above)
Comment (kdaud) [reply qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature:14]: @mherman22 the user story am referring to is 

```js
And a user clicks on the confirm button

```
changed to 

```js
And a user confirms the changes
```

The earlier phrasing of the user step is more clear.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 3]
```diff
@@ -20,95 +24,132 @@
...
-
-	@And("a user edits demographics")
+	@When("a user edits demographics")
 	public void editDemographics() {
-		testPatient.givenName = "Yuan";
-		testPatient.middleName = "Youn";
-		testPatient.familyName = "Ching";
 		testPatient.gender = "Male";
-		testPatient.birthDay = "23";
+		testPatient.birthDay = "21";
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:107]: any reason why changing the value of the birth day?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 4]
```diff
@@ -20,95 +24,132 @@
...
-		testPatient.city = "Kyambogo";
-		testPatient.state = "Central";
-		testPatient.country = "Uganda";
-		testPatient.postalCode = "256";
-		testPatient.phone = "0785456654";
+		testPatient.city = "Adidas Abbeba";
+		testPatient.state = "Adidas Abbeba";
+		testPatient.country = "Ethiopia";
+		testPatient.postalCode = "3822";
+		testPatient.phone = "aaaaaaaaa";
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:152]: is there a country where citizens telephone numbers are in alphabetic letters?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java (hunk 3 above)
Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:107]: Gonna erectify that

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java (hunk 4 above)
Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:152]: The alphabets are meant to test that the phone number only accepts numbers.

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:152]: > The alphabets are meant to test that the phone number only accepts numbers

This sounds to be xss vulnerability, @sherrif10 isn't ContactInfoTest supposed to be covered in [RATEST-308 ](https://issues.openmrs.org/browse/RATEST-308)?

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:152]: Nope thanks. they are quite different. with in this test , we are not handling vulnerability check the difference is between with in this test, we are not entering phone number with  any script element  tag for example <script>alert(0)</script>. which checks vulnerability. this test is supposed to be straight forward.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 5]
```diff
@@ -20,95 +24,132 @@
...
+		registrationEditSectionPage.enterFamilyName(testPatient.familyName + prefix);
 		registrationEditSectionPage.selectPatientGender(testPatient.gender);
 		registrationEditSectionPage.clickOnBirthdateLabel();
 		registrationEditSectionPage.enterBirthDay(testPatient.birthDay);
 		registrationEditSectionPage.selectBirthMonth(testPatient.birthMonth);
 		registrationEditSectionPage.enterBirthYear(testPatient.birthYear);
 	}
+	
+	@And ("a user confirms the changes")
+	public void confirmChanges() throws InterruptedException {
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:121]: take a close look at the method name

Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:121]: Care to eraborate?!

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java (hunk 2 above)
Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: i have also struggled to understand how tht figure came about here --> https://github.com/openmrs/openmrs-contrib-qaframework/blob/56c2c15f535d8a1e897cdcc0d2928b30620181f1/qaframework-legacy-tests/src/test/java/org/openmrs/contrib/qaframework/legacy/ContactInfoTest.java#L26

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: Thats fine . that uuid is simply a string concatenated via testData class.

Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: how about `new TestData.TestVisit(testPatient.uuid, TestData.getAVisitType(), getLocationUuid(homePage)).create();

Comment (sherrif10) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: Seems you are getting visitTypeUuid. isn't it part of the TestData attribute. Doe it work that way.

Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:38]: the test passes but anyway lemi keep the hardwired one for now

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 6]
```diff
@@ -20,95 +24,131 @@
...
-import org.openmrs.contrib.qaframework.page.RegistrationSummaryPage;
 
 public class EditDemographicsAndContactInfoSteps extends Steps {
 
+	private String prefix;
 	private TestData.PatientInfo testPatient;
+	private ActiveVisitsPage activeVisitsPage;
 	private RegistrationEditSectionPage registrationEditSectionPage;
-	private RegistrationSummaryPage registrationSummaryPage;
+	private FindPatientPage findPatientPage;
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:37]: do you need to declare the variable `findPatientPage`?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 7]
```diff
@@ -119,17 +159,14 @@ public void editContactInfo() {
 		registrationEditSectionPage.enterPostalCode(testPatient.postalCode);
 		registrationEditSectionPage.clickOnPhoneNumberEdit();
 		registrationEditSectionPage.enterPhoneNumber(testPatient.phone);
-	}
-
-	@And("a user clicks on the confirm button")
-	public void saveUpdatedPatientDetails() throws InterruptedException {
+		assertTrue(registrationEditSectionPage.getInvalidPhoneNumberNotification().contains("Must be a valid phone number (with +, -, numbers or parentheses)"));
+		registrationEditSectionPage.clearPhoneNumber();
+		registrationEditSectionPage.enterPhoneNumber("111111111");
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:164]: aren't these call methods supposed to be under `editContactInfo()`
```js
assertTrue(registrationEditSectionPage.getInvalidPhoneNumberNotification().contains("Must be a valid phone number (with +, -, numbers or parentheses)"));
registrationEditSectionPage.clearPhoneNumber();
registrationEditSectionPage.enterPhoneNumber("111111111");
```

**Review by sherrif10 — COMMENTED** [review]
Looks better using TestData.getaVisitType. btw isnt this a typo

**Comment by mherman22** [comment]
> 

actually its not

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java (hunk 6 above)
Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:37]: why not take advantage of inheritance?

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java (hunk 7 above)
Comment (mherman22) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:164]: not quite sure if i got your question clearly, care to elaborate or if im to answer as i understood it, the methods in question are under `editContactInfo()` as you have asked. you can a look-see at https://github.com/openmrs/openmrs-contrib-qaframework/blob/a67110b9cb88d15e26c3ed2ff8db2a8013d275b3/qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java#L146

Comment (kdaud) [reply qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:164]: could you take a close look at what saveUpdatedPatientDetails() is actually performing?

**Comment by sherrif10** [comment]
> > 
> 
> actually its not

True dat crossed checked it [here](https://github.com/openmrs/openmrs-contrib-qaframework/blob/master/qaframework-bdd-tests/src/main/java/org/openmrs/contrib/qaframework/helper/TestData.java#L68)

**Comment by sherrif10** [comment]
Addressed all the comments from @kdaud , Then we find this merged.

File: qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java [hunk 8]
```diff
@@ -9,7 +9,11 @@
  */
 package org.openmrs.contrib.qaframework.automation;
 
-import static org.junit.Assert.assertEquals;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.core.Is.is;
+import static org.junit.Assert.assertTrue;
+
+import org.apache.commons.lang3.RandomStringUtils;
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/java/org/openmrs/contrib/qaframework/automation/EditDemographicsAndContactInfoSteps.java:16]: align the import order with the existing one.

**Comment by mherman22** [comment]
cc: @sherrif10 @kdaud

File: qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature [hunk 9]
```diff
@@ -1,21 +1,27 @@
 Feature: Edit Demographics and contact info from Patient Dashboard
 
-  Background:
-    Given a user clicks on Edit Registration Information link from Patient dashboard
-    Then the system loads Registration Summary Page
-
   @selenium
   @patientDemographics
   Scenario: Edit Patient Demographics and Contact Info
```
Comment (kdaud) [inline qaframework-bdd-tests/src/test/resources/features/refapp-2.x/03-trackPatient/editDemographicsAndContactInfo.feature:5]: is this scenario covering editing contact information?

**Review by sherrif10 — APPROVED** [review]
Thanks @mherman22

**Pull request merged**